crowbar_framework/app/models/node_object.rb

```diff
@@ -1064,6 +1064,30 @@ def ssh_cmd(cmd)
     [200, {}]
   end
 
+  # Check for the presence of given file on the node
+  def file_exists?(file)
+    out = run_ssh_cmd("test -e #{file}")
+    out[:exit_code].zero?
+  end
+
+  # Waits until the background script finishes.
+  # We expect that the script generates two kinds of files to indicate success or failure
+  # Raises timeout exception if the waiting time exceedes 'seconds'
+  def wait_for_script_to_finish(seconds, ok_file, failed_file)
```

Comment:
Since it seems that `ok_file` and `failed_file` always have the identical path and prefix, why not just pass `script_name` instead?  Then you can calculate them inside this:

    base = "/var/lib/crowbar/upgrade/#{script_name}"
    ok_file = base + "-ok"
    failed_file = base + "-failed"

which eliminates duplication in the callers.